qa/vagrant/src/test/resources/packaging/scripts/modules.bash

```diff
@@ -0,0 +1,42 @@
+#!/bin/sh
+
+# This file contains some utilities to test the elasticsearch scripts,
+# the .deb/.rpm packages and the SysV/Systemd scripts.
+
+# WARNING: This testing file must be executed as root and can
+# dramatically change your system. It removes the 'elasticsearch'
+# user/group and also many directories. Do not execute this file
+# unless you know exactly what you are doing.
+
+# Licensed to Elasticsearch under one or more contributor
+# license agreements. See the NOTICE file distributed with
+# this work for additional information regarding copyright
+# ownership. Elasticsearch licenses this file to you under
+# the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+check_module() {
```

Comment:
Having to update the vagrant tests when we change a dependency is annoying. Originally I added the plugin test dependency checking to make sure we actually installed those files at all and that we could read the zip and stuff. I just don't trust mocks that much. OTOH I was pretty lenient with the file matching patterns. This has turned out to be a maintenance burden but I still stand by the idea behind the test.

I believe the bit that was the maintenance burden was around listing each plugin in the test case, not so much around the plugin's dependencies. I imagine we could generate the test case file with gradle now. When I originally wrote it we were on maven and I wasn't about to figure out how to do that. But gradle might be like 10 lines of build file.

I don't know that it'd alleviate the maintenance burden much to check fewer plugin files so I don't think that change is worth making. I still think we need to make sure we install every plugin on all the supported OSes.

So, maybe we do this as is and followup with a gradle change to generate the plugins test cases?